database/build.gradle

```diff
@@ -23,17 +23,16 @@ android {
 }
 
 dependencies {
-    compile "com.android.support:recyclerview-v7:$supportLibraryVersion"
-    // Needed to override play services
-    compile "com.android.support:support-v4:$supportLibraryVersion"
+    implementation "com.android.support:recyclerview-v7:$supportLibraryVersion"
+    implementation "com.android.support:support-v4:$supportLibraryVersion"
 
-    compile 'android.arch.lifecycle:runtime:1.0.0-alpha8'
-    compile 'android.arch.lifecycle:extensions:1.0.0-alpha8'
-    annotationProcessor 'android.arch.lifecycle:compiler:1.0.0-alpha8'
+    api 'android.arch.lifecycle:runtime:1.0.0-alpha9'
+    api 'android.arch.lifecycle:extensions:1.0.0-alpha9'
+    annotationProcessor 'android.arch.lifecycle:compiler:1.0.0-alpha9'
 
-    compile "com.google.firebase:firebase-database:$firebaseVersion"
+    api "com.google.firebase:firebase-database:$firebaseVersion"
```

Comment:
Do we really want to do this? This would require library users to match the version that we use. The auth module doesn't do this for its respective Firebase dependencies.

Reply:
I don't consider myself an expert on the new `3.0` stuff, but as far as I know you have to use `api` if you expose your dependencies in your public API.  Because we have classes that accept a `Query` or a `DatabaseReference`, this is our only option.

With auth, we completely hide all of the Firebase classes from our public API.  So we can get away with `implementation`.

Reply:
Darn. So much for non-transient dependencies. I wish this wasn't the case as we'd still have to upload the update the library for updates to the Firebase Database SDK to avoid a version mismatch. Or is there a way to still avoid this scenario?

Reply:
There may be, but I am not aware.